src/mqtt_client.rs

```diff
@@ -379,10 +397,9 @@ impl<
 
         // TODO: Generate event.
         if let Some(id) = publish.packet_id {
-            self.sm
-                .context_mut()
-                .session_state
-                .handle_publish(publish.qos, id, packet)?;
+            let context = self.sm.context_mut();
+            context.session_state.handle_publish(publish.qos, id, packet)?;
+            context.send_quota = context.send_quota.saturating_sub(1);
```

Comment:
Not sure if this should be saturating or not. Technically, it would be a defect in minimq if we tried to decrease beyond zero

Reply:
`checked_sub().unwrap()` then?